File: SkillHelper/src/main/java/com/skillhelper/api/ShareController.java
package com.skillhelper.api;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/share")
public class ShareController {
    @PostMapping("/add")
    public void AddShare() {

    }

    @PutMapping("/read")
    public void ReadShare() {

    }

    @DeleteMapping("/deleteMessages/{username}")
    public void DeleteMessages(@PathVariable String username) {

    }

    @GetMapping("/getAll/{username}")
    public void GetAll(@PathVariable String username) {

    }

    @GetMapping("/getById/{id}")
    public void GetById(@PathVariable long id) {
        
    }
}

File: SkillHelper/src/main/kotlin/com/skillhelper/api/ShareController.kt
package com.skillhelper.api

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/share")
class ShareController {
    @PostMapping("/add")
    fun addShare() {
    }

    @PutMapping("/read")
    fun readShare() {
    }

    @DeleteMapping("/deleteMessages/{username}")
    fun deleteMessages(@PathVariable username: String?) {
    }

    @GetMapping("/getAll/{username}")
    fun getAll(@PathVariable username: String?) {
    }

    @GetMapping("/getById/{id}")
    fun getById(@PathVariable id: Long) {
    }
}